### DISCUSSION TITLE
esm2 input sequence length limit

### DISCUSSION THREAD

**Discussion by ynuozhang** [facebookresearch/esm#486] [Q&A]
Hi,
I'm using the esm2 model to generate logits. I noticed that in the class `BatchConverter` in `data.py` https://github.com/facebookresearch/esm/blob/f8d030bc6495063046ee051b7e2b12eeebae7270/esm/data.py#L258 the default `truncation_seq_length` is set to None as an option. May I get some clarification here?
1. Is there technically no length restriction for the input sequences for esm2?
2. Because of the Rotary Position Embedding used?
3. RoPE has the long-term decay property, so it's better to run the model with sequences <= 1022 (truncate if necessary) like the `extract.py` shows?
